autoblocks/_impl/tracer.py

```diff
@@ -101,6 +123,70 @@ def start_span(self):
                 props["span_id"] = prev_span_id
             self.set_properties(props)
 
+    async def _evaluate_event(self, event: TracerEvent, evaluator: BaseEventEvaluator) -> Optional[Evaluation]:
+        """
+        Evaluates an event using a provided evaluator.
+        """
+        evaluation = None
+        if inspect.iscoroutinefunction(evaluator.evaluate_event):
+            try:
+                evaluation = await evaluator.evaluate_event(event=event)
+            except Exception as err:
+                return Evaluation(
+                    score=0,
+                    metadata={
+                        "error": "Evaluation raised an exception during execution",
+                        "errorDetail": f"{repr(err)}",
+                    },
+                )
+        else:
+            try:
+                ctx = contextvars.copy_context()
+                evaluation = await global_state.event_loop().run_in_executor(
+                    None,
+                    ctx.run,
+                    evaluator.evaluate_event,
+                    event,
+                )
+            except Exception as err:
+                return Evaluation(
+                    score=0,
+                    metadata={
+                        "error": "Evaluation raised an exception during execution",
+                        "errorDetail": f"{repr(err)}",
+                    },
+                )
+
+        return evaluation
+
+    async def _run_evaluators(
+        self, evaluators: List[BaseEventEvaluator], event: TracerEvent, max_evaluator_concurrency: int
+    ) -> List[ClickHouseEvaluationObject]:
+        if len(evaluators) == 0:
+            return []
+        try:
+            evaluations: List[Evaluation] = await gather_with_max_concurrency(
+                max_evaluator_concurrency,
+                [
+                    self._evaluate_event(
+                        event=event,
+                        evaluator=evaluator,
+                    )
+                    for evaluator in evaluators
+                ],
+            )
+            # Join evaluation results with their evaluator ids
+            # if results is not of type Evaluation add an evaluation
+            # with score 0 and metadata with the exception message
+            return [
+                ClickHouseEvaluationObject.to_json(event_evaluation=item, evaluator_external_id=evaluators[index].id)
```

Comment:
can you add a test w/ multiple evaluators and ensure they're matching up as you expect